holoviews/core/spaces.py

```diff
@@ -667,9 +680,19 @@ def clone(self, data=None, shared_data=True, new_type=None, *args, **overrides):
         """
         if data is None and shared_data:
             data = self.data
-        return super(UniformNdMapping, self).clone(overrides.pop('callback', self.callback),
-                                                   shared_data, new_type,
-                                                   *(data,) + args, **overrides)
+        clone = super(UniformNdMapping, self).clone(overrides.pop('callback', self.callback),
+                                                    shared_data, new_type,
+                                                    *(data,) + args, **overrides)
+
+        # Ensure the clone references this object to ensure
+        # stream sources are inherited
+        if clone.callback is self.callback:
+            clone.callback = self.callback.clone()
```

Comment:
It is at the top of the diff and I must have just skimmed over ``clone`` without registering it. Thanks for clarifying!